File: challenges/003/odd_integer.py
def find_it(seq):
    unique_integers = []

    for digit in seq:
        if digit not in unique_integers:
            unique_integers.append(digit)

    for int in unique_integers:
        count = 0

        for digit in seq:
            if int == digit:
                count += 1

        if count % 2 != 0:
            return int


"""
Simpler version
def find_it(seq):

    for int in seq:
        if seq.count(int) % 2 != 0:
            return int
"""

File: challenges/003/oddInteger.js
function findOdd(A) {

    let oddInteger = 0;
    const uniqueIntegers = [];

    for (const digit of A) {
        if (!uniqueIntegers.includes(digit))
            uniqueIntegers.push(digit)
    }

    for (const int of uniqueIntegers) {
        let integerCount = 0

        for (const digit of A) {
            if (int == digit) {
                integerCount ++;
            }
        }
        
        if (integerCount % 2 != 0) {
            oddInteger += int;
        }
    }

    console.log(oddInteger)
    return oddInteger;

}


// Simpler version usiung js builtin methods
/*
function findOdd(A) {
    return A.find((num) => A.filter((item) => item === num).length % 2 !== 0);
}
*/